internal/rpc/health/health.go

```diff
@@ -0,0 +1,45 @@
+package health
+
+import (
+	"github.com/NethermindEth/juno/internal/cairovm"
+	"go.uber.org/zap"
+
+	sync2 "github.com/NethermindEth/juno/internal/sync"
+
+	. "github.com/NethermindEth/juno/internal/log"
+)
+
+type Rpc struct {
+	synchronizer *sync2.Synchronizer
+	vm           *cairovm.VirtualMachine
+	logger       *zap.SugaredLogger
+}
+
+// New returns a new health rpc service.
+// notest
+func New(synchronizer *sync2.Synchronizer, vm *cairovm.VirtualMachine) *Rpc {
+	return &Rpc{
+		synchronizer: synchronizer,
+		vm:           vm,
+		logger:       Logger.Named("Health Check RPC"),
+	}
+}
+
+// NodeStatus returns the current node status.
+// notest
+func (r *Rpc) NodeStatus() (any, error) {
+	if r.synchronizer.Running && r.vm.Running() {
+		return Status{
+			Status:        "Healthy",
+			Message:       "Node running",
+			SyncingStatus: r.synchronizer.Status(),
+		}, nil
+	}
+	if !r.synchronizer.Running {
+		return nil, NodeNotSyncing
+	}
+	if !r.vm.Running() {
+		return nil, VMNotRunning
+	}
+	return nil, UnHealthy
+}
```

Comment:
Since this is a new package I feel like it is relatively self-contained. Therefore, writing tests should be easy. The logger should have been passed as a parameter to start moving away from the log singleton. This only creates further technical debt.